server/middlewares/auth.js

```diff
@@ -15,6 +16,39 @@ const validateToken = async (req, res, next) => {
   }
 };
 
-module.exports = {
-  validateToken,
-};
+exports.validateRegister = [
+  body("username")
+    .isLength(6)
+    .withMessage("Username should have at least six characters"),
+  body("password")
+    .isLength(6)
+    .withMessage("Password should have at least six characters"),
+  body("email").isEmail().withMessage("Email format is invalid"),
+  (req, res) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.status(400).json({
+        errors: errors.array(),
+      });
+    }
```

Comment:
I think that you are missing the next(). Otherwise, this request will not reach the controller

Reply:
i fix it 
now it should work.